be/src/exec/vectorized/schema_scan_node.cpp

```diff
@@ -174,64 +178,55 @@ Status SchemaScanNode::get_next(RuntimeState* state, ChunkPtr* chunk, bool* eos)
     const std::vector<SlotDescriptor*>& src_slot_descs = _schema_scanner->get_slot_descs();
     const std::vector<SlotDescriptor*>& dest_slot_descs = _dest_tuple_desc->slots();
 
-    *chunk = std::make_shared<vectorized::Chunk>();
-    if (nullptr == chunk->get()) {
+    bool scanner_eos = false;
+    int row_num = 0;
+
+    ChunkPtr chunk_src = std::make_shared<vectorized::Chunk>();
+    if (nullptr == chunk_src.get()) {
         return Status::InternalError("Failed to allocate new chunk.");
     }
-    for (auto& slot_desc : dest_slot_descs) {
-        if (!slot_desc->is_materialized()) {
-            continue;
-        }
+
+    for (auto& slot_desc : src_slot_descs) {
         ColumnPtr column = vectorized::ColumnHelper::create_column(slot_desc->type(), slot_desc->is_nullable());
-        (*chunk)->append_column(std::move(column), slot_desc->id());
+        chunk_src->append_column(std::move(column), slot_desc->id());
     }
-    bool scanner_eos = false;
-    int row_num = 0;
 
-    while (true) {
-        RETURN_IF_CANCELLED(state);
+    // convert src chunk format to dest chunk format to process where clause
+    ChunkPtr chunk_dst = std::make_shared<vectorized::Chunk>();
+    if (nullptr == chunk_dst.get()) {
+        return Status::InternalError("Failed to allocate new chunk.");
+    }
 
-        if (reached_limit()) {
-            _is_finished = true;
-            // if row_num is greater than 0, in this call, eos = false, and eos will be set to true
-            // in the next call
-            if (row_num == 0) {
-                *eos = true;
+    while (chunk_dst->is_empty()) {
+        while (row_num < runtime_state()->chunk_size()) {
+            if (reached_limit()) {
+                _is_finished = true;
+                // if row_num is greater than 0, in this call, eos = false, and eos will be set to true
+                // in the next call
+                if (row_num == 0) {
+                    *eos = true;
+                }
+                break;
```

Comment:
break should break out of `while (chunk_dst->is_empty()) {`

Reply:
No, needn't